module/zfs/abd.c

```diff
@@ -215,15 +221,37 @@ abd_fini(void)
 		kstat_delete(abd_ksp);
 		abd_ksp = NULL;
 	}
+}
+
+#else
 
-	kmem_cache_destroy(abd_chunk_cache);
-	abd_chunk_cache = NULL;
+struct page;
+#define	kpm_enable			1
+#define	abd_alloc_chunk() \
+	((struct page *)kmem_alloc(PAGESIZE, KM_SLEEP))
+#define	abd_free_chunk(chunk) 		kmem_free(chunk, PAGESIZE)
```

Comment:
Getting `PAGE_SIZE` wrong in user space shouldn't be terrible.  I could see it perhaps causing some memory alignment issues but I suspect that would be the worst of it.  That said I've never tried.

On the kernel side we should always be using `PAGE_SIZE` from the kernel headers.  I don't recall why we ever added the `#ifdef` around PAGESIZE, it seems like we'd want it to fail if `PAGESIZE` was ever defined.